scripts/generate_client_schema.py

```diff
@@ -0,0 +1,203 @@
+"""Generate the client schemas for the server."""
+
+import json
+from pathlib import Path
+
+from datamodel_code_generator import (
+    DataModelType,
+    Formatter,
+    PythonVersion,
+)
+from datamodel_code_generator.format import CodeFormatter
+from datamodel_code_generator.model import DataModel, get_data_model_types
+from datamodel_code_generator.parser.openapi import OpenAPIParser
+
+PROJECT_ROOT = Path(__file__).parent.parent
+CLIENT_DIR = PROJECT_ROOT / "lilypad/server/client"
+CLIENT_SCHEMAS_DIR = CLIENT_DIR / "schemas"
+CLIENT_SCHEMAS_INIT = CLIENT_SCHEMAS_DIR / "__init__.py"
+CLIENT_INIT = CLIENT_DIR / "__init__.py"
+SCHEMAS_FILE_HEADER = '"""The Schema models for the Lilypad {name} API."""'
+CLIENT_INIT_FILE_HEADER = '"""Client module for Lilypad server."""'
+SCHEMAS_INIT_FILE_HEADER = '"""The Schema models for the Lilypad API."""'
+
+OPENAPI_SCHEMAS_PATH = "#/components/schemas"
+
+FORMATTERS = [Formatter.RUFF_CHECK, Formatter.RUFF_FORMAT]
+
+ENCODING = "utf-8"
+
+
+V0_TARGET_MODELS = [
+    "GenerationPublic",
+    "OrganizationPublic",
+    "ProjectPublic",
+    "SpanPublic",
+    "GenerationCreate",
+    "Provider",
+]
+EE_V0_TARGET_MODELS = []
+
+
+class LilypadOpenAPIParser(OpenAPIParser):
```

Comment:
I implemented a slightly tricky solution to output only specific models. Since this project only deals with simple OpenAPI files, I believe it won't cause any issues.
If time permits, I'll add this feature to the mainstream branch.